src/code/stores/palette-store.ts

```diff
@@ -85,6 +89,57 @@ export const PaletteStore: PaletteStoreClass = Reflux.createStore({
     return this.updateChanges();
   },
 
+  /**
+   * Loads image collections from the JSON url defined at the url param "imageCollections".
+   * The JSON file may either be a collection, with a "title" and "images", or a list of
+   * collections under the key "collections".
+   *
+   * A bit of complexity is added to sort the final collections so that, regardless of how
+   * they are nested (e.g. collections or collections next to collections of images), when
+   * laid out in a tree we will sort in left-right order.
+   */
+  initializeCollections() {
+    this.collections = [];
+    if (urlParams.imageCollections) {
+
+      const collectionsOrder = [urlParams.imageCollections];
+
+      const collectionLoaded = (collection, url) => {
+        collection.url = url;
+        this.collections.push(collection);
+
+        this.collections.sort((c1, c2) => {
+          return collectionsOrder.indexOf(c1.url) - collectionsOrder.indexOf(c2.url);
+        });
+
+        this.updateChanges();
+      };
+
+      const loadCollection = (url) => {
+        log.info(`loading collection ${url}`);
+        return $.ajax({
+          url,
+          dataType: "json",
+          success: data => {
+            if (data.collections) {   // nested collection
+              // first replace our current url in collectionsOrder with our new list
+              collectionsOrder.splice(collectionsOrder.indexOf(url), 1, ...data.collections);
+              // then recursively load nested collections
+              data.collections.forEach(loadCollection);
```

Comment:
We have been using `lodash` for for loops on arrays based on older browsers not supporting that but I think enough time has passed that using the built in functions is fine now.

Reply:
Hmmm, `forEach` has been [supported since IE 9](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Browser_compatibility) (and Chrome 1, Firefox 1.5, etc.). Was that ever really a concern for us?